src/librustc/traits/project.rs

```diff
@@ -827,7 +827,9 @@ fn assemble_candidates_from_predicates<'cx, 'gcx, 'tcx, I>(
                predicate);
         match predicate {
             ty::Predicate::Projection(ref data) => {
-                let same_def_id = data.0.projection_ty.item_def_id == obligation.predicate.item_def_id;
+                let same_def_id = {
+                    data.0.projection_ty.item_def_id == obligation.predicate.item_def_id;
```

Comment:
Misplaced semicolon - you wouldn't have the block around such an indented line so just remove the braces.